plugin/pkg/admission/serviceplan/defaultserviceplan/admission_test.go

```diff
@@ -0,0 +1,196 @@
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/internalclientset/fake"
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers_generated/internalversion"
+)
+
+// newHandlerForTest returns a configured handler for testing.
+func newHandlerForTest(internalClient internalclientset.Interface) (admission.Interface, informers.SharedInformerFactory, error) {
+	f := informers.NewSharedInformerFactory(internalClient, 5*time.Minute)
+	handler, err := NewDefaultServicePlan()
+	if err != nil {
+		return nil, f, err
+	}
+	pluginInitializer := scadmission.NewPluginInitializer(internalClient, f, nil, nil)
+	pluginInitializer.Initialize(handler)
+	err = admission.Validate(handler)
+	return handler, f, err
+}
+
+// newFakeServiceCatalogClientForTest creates a fake clientset that returns a
+// ServiceClassList with the given ServiceClass as the single list item.
+func newFakeServiceCatalogClientForTest(sc *servicecatalog.ServiceClass) *fake.Clientset {
+	fakeClient := &fake.Clientset{}
+
+	scList := &servicecatalog.ServiceClassList{
+		ListMeta: metav1.ListMeta{
+			ResourceVersion: "1",
+		}}
+	scList.Items = append(scList.Items, *sc)
+
+	fakeClient.AddReactor("list", "serviceclasses", func(action core.Action) (bool, runtime.Object, error) {
+		return true, scList, nil
+	})
+	return fakeClient
+}
+
+// newInstance returns a new instance for the specified namespace.
+func newInstance(namespace string) servicecatalog.Instance {
+	return servicecatalog.Instance{
+		ObjectMeta: metav1.ObjectMeta{Name: "instance", Namespace: namespace},
+	}
+}
+
+// newServiceClass returns a new instance with the specified plans.
+func newServiceClass(name string, plans ...string) *servicecatalog.ServiceClass {
+	sc := &servicecatalog.ServiceClass{ObjectMeta: metav1.ObjectMeta{Name: name}}
+	for _, plan := range plans {
+		sc.Plans = append(sc.Plans, servicecatalog.ServicePlan{Name: plan})
+	}
+	return sc
+}
+
+func TestWithListFailure(t *testing.T) {
+	fakeClient := &fake.Clientset{}
+	fakeClient.AddReactor("list", "serviceclasses", func(action core.Action) (bool, runtime.Object, error) {
+		return true, nil, fmt.Errorf("simulated test failure")
+	})
+	handler, informerFactory, err := newHandlerForTest(fakeClient)
+	if err != nil {
+		t.Errorf("unexpected error initializing handler: %v", err)
+	}
+	informerFactory.Start(wait.NeverStop)
+
+	instance := newInstance("dummy")
+	instance.Spec.ServiceClassName = "foo"
+
+	err = handler.Admit(admission.NewAttributesRecord(&instance, nil, servicecatalog.Kind("Instance").WithVersion("version"), instance.Namespace, instance.Name, servicecatalog.Resource("instances").WithVersion("version"), "", admission.Create, nil))
+	if err == nil {
+		t.Errorf("unexpected success with no ServiceClasses.List succeeding")
+	} else if !strings.Contains(err.Error(), "not yet ready to handle request") {
```

Comment:
I can't express how much doing strcmp on error string like this in golang disturbs me :-(    nothing to change, just venting